node/core/av-store/src/lib.rs

```diff
@@ -110,62 +419,234 @@ impl AvailabilityStoreSubsystem {
 		let db = Database::open(&db_config, &path)?;
 
 		Ok(Self {
+			pruning_config: PruningConfig::default(),
 			inner: Arc::new(db),
 			metrics,
 		})
 	}
 
 	#[cfg(test)]
-	fn new_in_memory(inner: Arc<dyn KeyValueDB>) -> Self {
+	fn new_in_memory(inner: Arc<dyn KeyValueDB>, pruning_config: PruningConfig) -> Self {
 		Self {
+			pruning_config,
 			inner,
 			metrics: Metrics(None),
 		}
 	}
 }
 
-async fn run<Context>(subsystem: AvailabilityStoreSubsystem, mut ctx: Context)
+fn get_next_pov_pruning_time(db: &Arc<dyn KeyValueDB>) -> Option<NextPoVPruning> {
+	query_inner(db, columns::META, &NEXT_POV_PRUNING)
+}
+
+fn get_next_chunk_pruning_time(db: &Arc<dyn KeyValueDB>) -> Option<NextChunkPruning> {
+	query_inner(db, columns::META, &NEXT_CHUNK_PRUNING)
+}
+
+async fn run<Context>(mut subsystem: AvailabilityStoreSubsystem, mut ctx: Context)
 	-> Result<(), Error>
 where
 	Context: SubsystemContext<Message=AvailabilityStoreMessage>,
 {
 	let ctx = &mut ctx;
 	loop {
+		let pov_pruning_time = subsystem.maybe_prune_povs()?;
+		let chunk_pruning_time = subsystem.maybe_prune_chunks()?;
```

Comment:
It seems that we will do DB reads on every subsystem message in order to refresh these values. They are probably cached so it is fast, but maybe something to watch out for

Reply:
A comment pointing that out wouldn't harm, I think